pkg/gadgets/socket-collector/gadget.go

```diff
@@ -80,6 +80,10 @@ func (t *Trace) Operation(trace *gadgetv1alpha1.Trace,
 	case "start":
 		var pid uint32 = 0
 
+		if trace.Spec.Filter == nil {
```

Comment:
Thanks for handling this. I was adding this on coming PR for improving the filtering.